src/gc/gc.d

```diff
@@ -2033,6 +2208,13 @@ struct Gcx
         return 1;
     }
 
+    static struct ScanRange
+    {
+        void* pbot;
+        void* ptop;
+        Pool* pool;
```

Comment:
I checked the disassembly: it's all inlined AFAICT. So the struct needs 50% more memory (there is no padding), but there is probably not a lot we can do here. The non-precise version might use the Range struct though, so we could templatize ToScanStack on that type.
Indexing might be a little more expensive than with a struct size of 8/16, so we could also use a pointer to walk the stack.